build.go

```diff
@@ -122,6 +122,9 @@ type BuildOptions struct {
 	// Only trust builders from reputable sources.
 	TrustBuilder bool
 
+	// The build and stack user's group id must be override
+	OverrideGroupID bool
```

Comment:
I'm not particularly a fan of having two fields that must be used in unison. I know we opted out of using `-1` to signify "don't override" from a user's perspective but I think it's perfectly fine to use `-1` internally. Then we don't have to worry about two state variables.

IOW, we can keep the current logic in `cmd` package where it looks for the presence of the `gid` flag but instead of using `OverrideGroupID`, it updates `GroupID` to `-1` to signify that we shouldn't override it on the image or provide it to the lifecycle. That should achieve the same behaviour right?

Reply:
@jromero Yeap, I agreed, we can keep the same logic using that approach. Good one!